common/types/client.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+import (
+	"context"
+	"math/big"
+)
+
+type Client[CHAINID any, ADDR comparable, BLOCK any, TX any, HASH comparable, TXRECEIPT any, EVENT any, EVENTOPS any] interface {
+	ChainID() (CHAINID, error)
+
+	// account
+	BalanceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (*big.Int, error)
+	TokenBalance(ctx context.Context, accountAddress ADDR, tokenAddress ADDR) (*big.Int, error)
+	SequenceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (uint64, error)
```

Comment:
These comments are human readable in code, but they won't make sense on the rendered godoc, since they are only attached to the first method of each group. Another option would be to define each sub-inteface separately, then embed them here:
```suggestion
	ClientAccounts
```
```go
// ClientAccounts ...
type ClientAccounts[ADDR comparable] interface {
	BalanceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (*big.Int, error)
	TokenBalance(ctx context.Context, accountAddress ADDR, tokenAddress ADDR) (*big.Int, error)
	SequenceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (uint64, error)
}
```